### PR TITLE
Qt server in agent

### PR DESCRIPTION
**Pull request by wkucjas**
Resolution for Agent: listen for connections #14

### CODE REVIEW

File: src/agent/Server.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#include "Server.h"
+#include <qnetworkinterface.h>
```
Comment (Kicer86) [inline src/agent/Server.cpp]: #include <QNetworkInterface>
Same for datastream and tcp socket

File: src/agent/Server.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    connect(&tcpServer, &QTcpServer::newConnection, this, &Server::SendData);
+}
+
+bool Server::Init()
+{
+        if (!tcpServer.listen(QHostAddress::Any, RDHMPort)) {
+            return false;
+        }
+
+        QString ipAddress;
```
Comment (Kicer86) [inline src/agent/Server.cpp:20]: QTcpServer::serverAddress()  ?

File: src/agent/Server.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+{
+    Q_OBJECT
+
+public:
+    Server(QObject* parent = nullptr);
+    bool Init();
+private:
+    void SendData();
+
+   QTcpServer tcpServer;
```
Comment (Kicer86) [inline src/agent/Server.h]: m_

**Review by Kicer86 — CHANGES_REQUESTED** [review]
.

File: src/agent/Server.cpp (hunk 2 above)
Comment (wkucjas) [reply src/agent/Server.cpp:20]: Nope. This method returns 0.0.0.0 always

File: src/agent/Server.h (hunk 3 above)
Comment (wkucjas) [reply src/agent/Server.h]: Fixed in next branch but ok :)

**Pull request merged**